src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/BlobContentId.cs

```diff
@@ -22,17 +22,27 @@ public BlobContentId(Guid guid, uint stamp)
         }
 
         public BlobContentId(ImmutableArray<byte> id)
-            : this(ImmutableByteArrayInterop.DangerousGetUnderlyingArray(id)!)
         {
+            if (id.IsDefault)
+            {
+                Throw.ArgumentNull(nameof(id));
+            }
+
+            (Guid, Stamp) = Initialize(id.AsSpan());
         }
 
-        public unsafe BlobContentId(byte[] id)
+        public BlobContentId(byte[] id)
         {
             if (id is null)
             {
                 Throw.ArgumentNull(nameof(id));
             }
 
+            (Guid, Stamp) = Initialize(id);
+        }
+
+        private static unsafe (Guid guid, uint stamp) Initialize(ReadOnlySpan<byte> id)
```

Comment:
Do we need a reference to ValueTuple package for .NET Framework?
```
  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">
    <PackageReference Include="System.ValueTuple" Version="$(SystemValueTupleVersion)" />
  </ItemGroup>
```

Reply:
Not anymore, I refactored the code.